src/test/java/com/google/sps/ContextServletTest.java

```diff
@@ -36,25 +45,44 @@
 public class ContextServletTest {
   private static final String URL_PARAMETER = "url";
   private static final String TEST_URL = "WkZ5e94QnWk";
-  private static final String REQUEST_INFO = "snippet,contentDetails,statistics";
+  private static final String TEST_TITLE = "Test Title";
+  private static final String TEST_CHANNEL = "Test Channel";
+  private static final int NUM_LIKES_VAL = 10;
+  private static final int NUM_DISLIKES_VAL = 1;
 
   @Test
-  public void testContextGet() throws ServletException, IOException, GeneralSecurityException {
-    ContextServlet contextServlet = new ContextServlet();
+  public void testContextGet() throws IOException, GeneralSecurityException {
+    // Test the doGet() method to retrieve youtube vidoe context with mocked service
+    ContextServlet contextServlet = spy(ContextServlet.class);
     HttpServletRequest mockedRequest = mock(HttpServletRequest.class);
     HttpServletResponse mockedResponse = mock(HttpServletResponse.class);
-    when(mockedRequest.getParameter(URL_PARAMETER)).thenReturn(TEST_URL);
     StringWriter stringWriter = new StringWriter();
     PrintWriter writer = new PrintWriter(stringWriter);
+    VideoSnippet mockedVideoSnippet =
+        new VideoSnippet()
+            .setPublishedAt(new DateTime(new Date()))
+            .setTitle(TEST_TITLE)
+            .setChannelTitle(TEST_CHANNEL);
+    VideoStatistics mockedVidoeStatistics =
+        new VideoStatistics().setLikeCount(BigInteger.TEN).setDislikeCount(BigInteger.ONE);
+    Video mockedVideo =
+        new Video().setSnippet(mockedVideoSnippet).setStatistics(mockedVidoeStatistics);
+    VideoListResponse mockedVideoList =
+        new VideoListResponse().setItems(Collections.singletonList(mockedVideo));
+    when(mockedRequest.getParameter(URL_PARAMETER)).thenReturn(TEST_URL);
+    when(contextServlet.generateYouTubeRequest(TEST_URL)).thenReturn(mockedVideoList);
     when(mockedResponse.getWriter()).thenReturn(writer);
-    contextServlet.doGet(mockedRequest, mockedResponse);
-    verify(mockedRequest, atLeast(1))
-        .getParameter(URL_PARAMETER); // only if you want to verify username was called...
-    Assert.assertTrue(stringWriter.toString().contains("Fuck Donald Trump"));
-    Assert.assertTrue(stringWriter.toString().contains("\"videoAuthor\":\"WORLDSTARHIPHOP\""));
-    Assert.assertTrue(stringWriter.toString().contains("numLikes"));
-    Assert.assertTrue(stringWriter.toString().contains("numDislikes"));
-    Assert.assertTrue(
-        stringWriter.toString().contains("\"publishDate\":\"2016-04-18T17:01:23.000Z\""));
+    try {
+      contextServlet.doGet(mockedRequest, mockedResponse);
+    } catch (ServletException e) {
```

Comment:
You can make this test more compact by adding the `ServletException` to the `throws` clause for this test.